core/src/connection.rs

```diff
@@ -21,9 +21,10 @@
 use crate::multiaddr::{Multiaddr, Protocol};
 
 /// The endpoint roles associated with a peer-to-peer communication channel.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Default)]
 pub enum Endpoint {
     /// The socket comes from a dialer.
+    #[default]
```

Comment:
I don't see why `Endpoint::Dialer` should be the default. How is that intuitive? In my eyes `Dialer` and `Listener` are equal in the sense that neither is more common. Quite the opposite, given that each connection has `Dialer` side and a `Listener` side, they should be mostly equal.